recipes/libcusolver/meta.yaml

```
@@ -0,0 +1,104 @@
+{% set name = "libcusolver" %}
+{% set version = "11.4.2.57" %}
+{% set platform = "linux-x86_64" %}  # [linux64]
+{% set platform = "linux-ppc64le" %}  # [ppc64le]
+{% set platform = "linux-sbsa" %}  # [aarch64]
+{% set platform = "windows-x86_64" %}  # [win]
+{% set extension = "tar.xz" %}  # [not win]
+{% set extension = "zip" %}  # [win]
+{% set major_version = version.split(".")[0] %}
+{% set minor_version = version.split(".")[0] %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://developer.download.nvidia.com/compute/cuda/redist/{{ name }}/{{ platform }}/{{ name }}-{{ platform }}-{{ version }}-archive.{{ extension }}
+  sha256: c9460db500d0c7d14debb2c8e0d88fc26a39197363f1b000ccf760866f636f12  # [linux64]
+  sha256: bf3c9a090652d011aafe2190c6f26fe108194cf8512d3c571e406054740168bb  # [ppc64le]
+  sha256: 0ca46b73520657f96f921711c05af7053bd398328d510aecddfff2521c000e91  # [aarch64]
+  sha256: c09836df1f43a3496f2b61ad8827af662ace298508869564a40c30d909ba550f  # [win]
+
+build:
+  number: 0
+  skip: true  # [osx]
+
+requirements:
+  build:
+    - {{ compiler("c") }}
+    - {{ compiler("cxx") }}
+
+outputs:
+  - name: libcusolver
+    requirements:
+      run:
+        - libcublas =12.*
+        - libnvjitlink =12.*
```

nit: 
```suggestion
        - libnvjitlink >=12.0,<13
```
as per [CUDA 12.0 Compiler Support for Runtime LTO Using nvJitLink Library](https://developer.nvidia.com/blog/cuda-12-0-compiler-support-for-runtime-lto-using-nvjitlink-library/):
> Statically or dynamically link the highest version of nvJitLink library from the major release.

---

Is there a way we can do this without needing to know the next version? It is likely a safe bet here, but I'd like to remove the guess work on things out of our control if possible.

---

We can certainly set up some Jinja variables, but it is a requirement of nvJitLink to be compatible with the CTK major release IIUC. Ultimately, this needs to track back which CTK major release it belongs to. I see no other way based on my current understanding 😞

---

TODO: Beter to pin `cuda-version` instead (but how?)